src/WPF/WPF.Viewer/MainWindow.xaml.cs

```diff
@@ -227,6 +229,39 @@ private async Task SelectSample(SampleInfo selectedSample)
             SetContainerDimensions();
         }
 
+        private static void Sample_Unloaded(object sender, RoutedEventArgs e)
+        {
+            // Explicit cleanup of the Map and SceneView instead of waiting for garbage collector can help when
+            // lots of geoviews are being opened and closed
+            foreach (var geoView in TreeWalker<GeoView>(sender as UIElement))
+            {
+                if (geoView is MapView mapView)
+                {
+                    mapView.Map = null;
+                    if (mapView.LocationDisplay != null) mapView.LocationDisplay.IsEnabled = false;
+                }
+                else if (geoView is SceneView sceneView) sceneView.Scene = null;
+            }
+        }
+
+        private static IEnumerable<T> TreeWalker<T>(UIElement root)
+        {
+            if (root != null)
+            {
+                if (root is T t)
+                    yield return t;
+                else if (root is UIElement it)
+                {
+                    for (int i = 0; i < VisualTreeHelper.GetChildrenCount(it); i++)
+                    {
+                        var e = VisualTreeHelper.GetChild(it, 0);
```

Comment:
Did you mean to get the `i`th child, instead of always getting the first child?
```suggestion
                        var e = VisualTreeHelper.GetChild(it, i);
```